src/dex/algebra/algebra-pool.ts

```diff
@@ -227,6 +228,123 @@ export class AlgebraEventPool extends StatefulEventSubscriber<PoolState> {
+        callData: this.stateMultiContract.methods
+          .getFullStateWithoutTicks(
+            this.factoryAddress,
+            this.token0,
+            this.token1,
+            0,
+            0,
+          )
+          .encodeABI(),
+        decodeFunction: decodeStateMultiCallResultWithRelativeBitmaps,
+      },
+    ];
+
+    const [resBalance0, resBalance1, stateWithoutTicksAndTickBitmap] =
+      await this.dexHelper.multiWrapper.tryAggregate<
+        bigint | DecodedStateMultiCallResultWithRelativeBitmaps
+      >(
+        false,
+        balancesAndGlobalStateCalldata,
+        blockNumber,
+        this.dexHelper.multiWrapper.defaultBatchSize,
+        false,
+      );
+
+    assert(stateWithoutTicksAndTickBitmap.success, 'Pool does not exist');
+
+    const [balance0, balance1, _stateWithoutTicksAndTickBitmap] = [
+      resBalance0.returnData,
+      resBalance1.returnData,
+      stateWithoutTicksAndTickBitmap.returnData,
+    ] as [bigint, bigint, DecodedStateMultiCallResultWithRelativeBitmaps];
+
+    const {
+      globalState: { tick },
+    } = _stateWithoutTicksAndTickBitmap;
+    const currentBitmapIndex = int16(
+      (BigInt(tick) / Constants.TICK_SPACING) >> 8n,
+    );
+
+    const buffer = this.getBitmapRangeToRequest();
+    const startBitMapIndex = currentBitmapIndex - buffer;
+    const endBitMapIndex = currentBitmapIndex + buffer;
+    const allBitMapIndices = Array.from(
+      { length: 2 * Number(buffer) + 1 },
+      () => startBitMapIndex + endBitMapIndex,
+    );
+
+    const ticksAndBitMaps = await Promise.all(
+      allBitMapIndices.map(relativeTick => {
+        return this.stateMultiContract.methods
+          .getAdditionalBitmapWithTicks(
+            this.factoryAddress,
+            this.token0,
+            this.token1,
+            relativeTick,
+            relativeTick,
+          )
+          .call() as [
+          TickBitMapMappingsWithBigNumber[],
+          TickInfoMappingsWithBigNumber[],
+        ];
+      }),
+    );
+
+    const _state: DecodedStateMultiCallResultWithRelativeBitmaps = {
+      ..._stateWithoutTicksAndTickBitmap,
+      tickBitmap: ticksAndBitMaps.flatMap(v => v[0]),
+      ticks: ticksAndBitMaps.flatMap(v => v[1]),
+    };
+
+    return [balance0, balance1, _state];
+  }
+
+  async _fetchInitStateMultiStrategies(
+    blockNumber: number,
+  ): Promise<[bigint, bigint, DecodedStateMultiCallResultWithRelativeBitmaps]> {
+    try {
+      return await this._fetchPoolStateSingleStep(blockNumber);
+    } catch (e) {
+      if (e instanceof Error && e.message.includes('Pool does not exist'))
+        throw e;
+
+      if (this.dexHelper.config.data.network != Network.ZKEVM) throw e;
+
+      return this._fetchPoolStateMultiStep(blockNumber);
+    }
+  }
+
+  async generateState(blockNumber: number): Promise<Readonly<PoolState>> {
+    const [balance0, balance1, _state] =
+      await this._fetchInitStateMultiStrategies(blockNumber);
```

Comment:
As far as I understand, this function might fail, if we are on zkEVM and multi step multicall fails for some reason. In that case, I don't see any error handling for generate state. Maybe better to consider try catch for that situation [here](https://github.com/paraswap/paraswap-dex-lib/blob/bcac90b3f3c3817c8461eeb7573284f024ff246c/src/dex/algebra/algebra-pool.ts#L114) or [here](https://github.com/paraswap/paraswap-dex-lib/blob/bcac90b3f3c3817c8461eeb7573284f024ff246c/src/dex/algebra/algebra-pool.ts#L340) or [here](https://github.com/paraswap/paraswap-dex-lib/blob/bcac90b3f3c3817c8461eeb7573284f024ff246c/src/dex/algebra/algebra-pool.ts#L303) ?

Reply:
This case the status quo basically. When generateState fails, it's caught and state stays at init value (null)

Reply:
Ah, thanks. Found it. Here it is caught:
https://github.com/paraswap/paraswap-api/blob/37f6c026612fc36f834554a22ef363e8ae1c99f1/src/lib/block-manager.ts#L392